languages/python/polar/query.py

```diff
@@ -67,10 +67,17 @@ def run(self):
 
     def handle_make_external(self, data):
         id = data["instance_id"]
-        cls_name = data["instance"]["tag"]
-        fields = data["instance"]["fields"]["fields"]
-        fields = {k: self.host.to_python(v) for k, v in fields.items()}
-        self.host.make_instance(cls_name, fields, id)
+        constructor = data["constructor"]["value"]
+        if "InstanceLiteral" in constructor:
+            cls_name = constructor["InstanceLiteral"]["tag"]
+            fields = constructor["InstanceLiteral"]["fields"]["fields"]
+            initargs = {k: self.host.to_python(v) for k, v in fields.items()}
+        elif "Call" in constructor:
+            cls_name = constructor["Call"]["name"]
+            initargs = [self.host.to_python(arg) for arg in constructor["Call"]["args"]]
+        else:
+            raise PolarApiException("Bad constructor")
+        self.host.make_instance(cls_name, initargs, id)
```

Comment:
Ah okay I see, so construction can still be done by instance literal.

Reply:
Right.